### PR TITLE
Change https://nuttx.apache.org/download/ to https://www.apache.org/dyn/closer.lua

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
follow the new requirements from the mentor.

## Impact
The download link is difference.

### CODE REVIEW

File: _releases/9.0.0.md [hunk 1]
```diff
@@ -7,9 +7,9 @@ date: 2020-05-11
 summary: >
     Release v9.0.0
 
-artifact-root: "https://archive.apache.org/dist/incubator/nuttx/9.0.0"
```
Comment (btashton) [inline _releases/9.0.0.md:10]: The archive links should _not_ be changed. Only the download ones.

**Review by btashton — CHANGES_REQUESTED** [review]

File: _releases/9.0.0.md (hunk 1 above)
Comment (xiaoxiang781216) [reply _releases/9.0.0.md:10]: Done.

File: _releases/9.0.0.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ summary: >
 
 artifact-root: "https://archive.apache.org/dist/incubator/nuttx/9.0.0"
 checksum-root: "https://archive.apache.org/dist/incubator/nuttx/9.0.0"
-key-file: "https://downloads.apache.org/incubator/nuttx/KEYS"
+key-file: "https://www.apache.org/dyn/closer.lua/incubator/nuttx/KEYS"
```
Comment (btashton) [inline _releases/9.0.0.md:12]: Keys and signatures also _must_ be linked directly to downloads and not via the lua script see https://infra.apache.org/release-download-pages.html

**Review by btashton — CHANGES_REQUESTED** [review]

**Review by btashton — APPROVED** [review]

File: _releases/9.0.0.md (hunk 2 above)
Comment (xiaoxiang781216) [reply _releases/9.0.0.md]: Done.

**Review by Apache9 — APPROVED** [review]
+1

**Pull request merged**